src/lib/core/CASEAuthTag.h

```diff
@@ -56,6 +57,30 @@ struct CATValues
         }
         return false;
     }
+
+    static constexpr size_t kSerializedLength = kMaxSubjectCATAttributeCount * sizeof(CASEAuthTag);
+    typedef uint8_t Serialized[kSerializedLength];
+
+    CHIP_ERROR Serialize(Serialized & outSerialized) const
+    {
+        for (size_t i = 0; i < kMaxSubjectCATAttributeCount; i++)
+        {
+            uint32_t littleEndianCAT = Encoding::LittleEndian::HostSwap32(values[i]);
+            memcpy(&outSerialized[i * sizeof(littleEndianCAT)], &littleEndianCAT, sizeof(littleEndianCAT));
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR Deserialize(const Serialized inSerialized)
+    {
+        for (size_t i = 0; i < kMaxSubjectCATAttributeCount; i++)
+        {
+            uint32_t littleEndianCAT;
+            memcpy(&littleEndianCAT, &inSerialized[i * sizeof(littleEndianCAT)], sizeof(littleEndianCAT));
+            values[i] = Encoding::LittleEndian::HostSwap32(littleEndianCAT);
```

Comment:
Why not use Get32/Read32, without needing to memcpy?

Reply:
updated